server/server.js

```diff
@@ -56,27 +56,75 @@ app.middleware('parse', bodyParser.urlencoded({
   extended: true
 }));
 
-// The access token is only available after boot
-app.middleware('auth', loopback.token({
-  model: app.models.accessToken,
-  currentUserLiteral: 'me'
-}));
 
-app.use(loopback.token());
+
+// Establish current user from access token
+app.use(loopback.context());
+app.use(function setCurrentUser(req, res, next) {
+    if (!req.accessToken) {
+        return next();
+    }
+    app.models.Person.findById(req.accessToken.userId, function(err, user) {
+        if (err) {
+            return next(err);
+        }
+        if (!user) {
+            return next(new Error('No user with this access token was found.'));
+        }
+
+        // NOTE: This section is pulled from passport-configurator.js
+        // sorry.
+        user.identities(function(err, identities) {
+          user.profiles = identities;
+          user.credentials(function(err, accounts) {
+            user.accounts = accounts;
+
+            // Set this user on the current context
+            var loopbackContext = loopback.getCurrentContext();
+            if (loopbackContext) {
+                req.accessToken.currentUser = user;
+                loopbackContext.set('currentUser', user);
+            }
+            req.user = user;
+            // ^ didn't see examples doing this, but couldn't find a better way
+
+            next();
+
+          });
+        });
+
+
+    });
+});
+
 
 app.middleware('session:before', loopback.cookieParser(app.get('cookieSecret')));
 app.middleware('session', loopback.session({
   secret: 'kitty',
   saveUninitialized: true,
   resave: true
 }));
-passportConfigurator.init();
+
+// REF: http://blog.digitopia.com/tokens-sessions-users/
+// use loopback.token middleware on all routes
+// setup gear for authentication using cookie (access_token)
+// Note: requires cookie-parser (defined in middleware.json)
+app.use(loopback.token({  
+  model: app.models.accessToken,
+  currentUserLiteral: 'me',
```

Comment:
I note that `currentUserLiteral` is set here and also in `authentication.js`. That might be necessary, as I think the auth one is for the REST API, while this one is for general express routing.